File: swarmsolver-backend/src/main/java/ai/swarmsolver/backend/app/tool/dto/TooDescriptionDTO.java
package ai.swarmsolver.backend.app.tool.dto;

import ai.swarmsolver.backend.app.tool.model.ParameterType;
import lombok.Builder;
import lombok.Data;

import java.util.List;

@Builder
@Data
public class TooDescriptionDTO {
    private final String name;
    private final String description;
    private final List<ToolParameterDTO> parameters;
    private final ParameterType returnType;
    private final String returnTypeDetails;
}

File: swarmsolver-frontend/src/app/api/models/too-description-dto.ts
/* tslint:disable */
/* eslint-disable */
import { ToolParameterDto } from '../models/tool-parameter-dto';
export interface TooDescriptionDto {
  description?: string;
  name?: string;
  parameters?: Array<ToolParameterDto>;
  returnType?: 'SOURCE_CODE' | 'TEXT';
  returnTypeDetails?: string;
}
